File: java/src/main/java/abstract_factory/product/part/CityVehiclePartFactory.java
package abstract_factory.product.part;

import java.util.List;

public class CityVehiclePartFactory implements VehiclePartFactory {
    @Override
    public Engine createEngine() {
        return Engine.HYBRID;
    }

    @Override
    public Tire createTire() {
        return Tire.NORMAL;
    }

    @Override
    public List<Option> createOptions() {
        return List.of(Option.NAVIGATION, Option.CRUISE_CONTROL);
    }
}

File: src/main/kotlin/abstract_factory/product/part/CityVehiclePartFactory.kt
package abstract_factory.product.part

class CityVehiclePartFactory : VehiclePartFactory {
    override fun createEngine(): Engine {
        return Engine.HYBRID
    }

    override fun createTire(): Tire {
        return Tire.NORMAL
    }

    override fun createOptions(): List<Option> {
        return listOf(Option.NAVIGATION, Option.CRUISE_CONTROL)
    }

}